modules/pxp-module-puppet

```diff
@@ -91,12 +95,9 @@ def get_result_from_report(exitcode, config, start_time)
                              "#{last_run_report} could not be loaded: #{e}")
   end
 
-  run_result = last_run_result(exitcode)
-
-  if exitcode != 0
-    run_result = make_error_result(exitcode, Errors::NonZeroExit,
-                                   "Puppet agent exited with a non 0 exitcode")
-  end
+  run_result = exitcode == 0 ?  last_run_result(exitcode)
+    : make_error_result(exitcode, Errors::NonZeroExit,
```

Comment:
Use of the ternary here over multiple lines is really hard to read.  If we're going to touch this for clarity I'd suggest a more explicit structure.

```
if exitcode == 0
  run_result = last_run_result(exitcode)
else
  run_result = make_error_result(exitcode, Errors::NonZeroExit,  "Puppet agent exited with a non 0 exitcode")
end
```